src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -0,0 +1,111 @@
+#include "remote_control/resource_allocation_manager_impl.h"
+#include "application_manager/application.h"
+
+namespace remote_control {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "RemoteControlModule")
+
+ResourceAllocationManagerImpl::ResourceAllocationManagerImpl(
+    RemotePluginInterface& rc_plugin)
+    : current_access_mode_(AccessMode::AUTO_ALLOW)
+    , active_call_back_(NULL)
+    , rc_plugin_(rc_plugin) {}
+
+AcquireResult::eType ResourceAllocationManagerImpl::AcquireResource(
+    const std::string& module_type, uint32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const application_manager::ApplicationSharedPtr aquiring_app =
+      rc_plugin_.service()->GetApplication(app_id);
+  if (!aquiring_app) {
+    LOG4CXX_WARN(logger_, "App with app_id " << app_id << "does not exist!");
+    return AcquireResult::IN_USE;
+  }
+
+  const AllocatedResources::const_iterator allocated_it =
+      allocated_resources_.find(module_type);
+  if (allocated_resources_.end() == allocated_it) {
+    allocated_resources_[module_type] = app_id;
+    LOG4CXX_DEBUG(logger_,
+                  "Resource is not acquired yet. "
+                  "Allow "
+                      << app_id << " aquiring " << module_type);
+    return AcquireResult::ALLOWED;
+  }
+
+  const mobile_apis::HMILevel::eType aquiring_app_hmi_level =
+      aquiring_app->hmi_level();
+  if (mobile_apis::HMILevel::HMI_FULL != aquiring_app_hmi_level) {
+    LOG4CXX_DEBUG(logger_,
+                  "Aquiring resources not alllowed in "
+                      << aquiring_app_hmi_level << " hmi level. "
+                      << "Disallow " << app_id << " aquiring " << module_type);
+    return AcquireResult::IN_USE;
+  }
+
+  switch (current_access_mode_) {
+    case AccessMode::AUTO_DENY: {
+      LOG4CXX_DEBUG(logger_,
+                    "Current access_mode is AUTO_DENY. "
+                    "Disallow "
+                        << app_id << " aquiring " << module_type);
+      return AcquireResult::IN_USE;
+    }
+    case AccessMode::ASK_DRIVER: {
+      LOG4CXX_DEBUG(logger_,
+                    "Current access_mode is ASK_DRIVER. "
+                    "Driver confirm required for allowing "
```

Comment:
@LuxoftAKutsan Driver confirm required for allowing " - > Driver confirmation required to allow "